wp-content/themes/core/components/blocks/gallery_slider/Gallery_Slider_Controller.php

```diff
@@ -0,0 +1,305 @@
+	public function get_content_classes(): string {
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_header_args(): array {
+		if ( empty( $this->title ) && empty( $this->description ) ) {
+			return [];
+		}
+
+		return [
+			Content_Block_Controller::TAG     => 'header',
+			Content_Block_Controller::LAYOUT  => Content_Block_Controller::LAYOUT_CENTER,
+			Content_Block_Controller::TITLE   => $this->get_title(),
+			Content_Block_Controller::CONTENT => $this->get_content(),
+			Content_Block_Controller::CTA     => $this->get_cta(),
+			Content_Block_Controller::CLASSES => [
+				'c-block__content-block',
+				'c-block__header',
+				'b-gallery-slider__header'
+			],
+		];
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_title(): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::TAG     => 'h2',
+			Text_Controller::CLASSES => [
+				'c-block__title',
+				'b-gallery-slider__title',
+				'h3',
+			],
+			Text_Controller::CONTENT => $this->title ?? '',
+		] );
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_content(): Deferred_Component {
+		return defer_template_part( 'components/container/container', null, [
+			Container_Controller::CLASSES => [
+				'c-block__description',
+				'b-gallery-slider__description',
+				't-sink',
+				's-sink',
+			],
+			Container_Controller::CONTENT => $this->description ?? '',
+		] );
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_cta(): Deferred_Component {
+		$cta = wp_parse_args( $this->cta, [
+			'content' => '',
+			'url'     => '',
+			'target'  => '',
+		] );
+
+		return defer_template_part( 'components/link/link', null, [
+			Link_Controller::URL     => $cta['url'],
+			Link_Controller::CONTENT => $cta['content'] ?: $cta['url'],
+			Link_Controller::TARGET  => $cta['target'],
+			Link_Controller::CLASSES => [
+				'c-block__cta-link',
+				'a-btn',
+				'a-btn--has-icon-after',
+				'icon-arrow-right'
+			],
+		] );
+	}
+
+	/**
+	 * @return string
+	 */
+	protected function get_slider_options(): string {
+		$args = [
+			'preloadImages'  => "true",
+			'lazy'           => "true",
+			'keyboard'       => "true",
+			'grabCursor'     => "true",
+			'slidesPerView'  => 3,
+			'centeredSlides' => "true",
+			'spaceBetween'   => 40,
+		];
+
+		return json_encode( $args );
+	}
+
+	/**
+	 * Get the Slider
+	 *
+	 * @return array
+	 */
+	public function get_slider_args(): array {
+		$main_attrs = [];
+
+		$main_attrs['data-swiper-options'] = $this->get_slider_options();
+
+		$slider = [
+			Slider_Controller::SLIDES     => $this->get_slides(),
+			Slider_Controller::MAIN_ATTRS => $main_attrs,
+			Slider_Controller::CLASSES    => [ 'b-gallery-slider__slider', 'c-slider--arrows-below' ],
+		];
+
+		return $slider;
+	}
+
+	/**
+	 * Get Slides
+	 *
+	 * @return array
+	 */
+	public function get_slides(): array {
```

Comment:
@mmcachran can I get your eyes on this? I'm sure there's a more eloquent way of returning the slide markup. I used the Gallery Grid block as a base when I started this one and it returns an array_map. Sort of seemed like overkill here but also seems like a BE decision.